jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java

```diff
@@ -208,12 +217,23 @@ private static String prefix(String prefix) {
     return prefix != null ? String.format("%s-", prefix) : "";
   }
 
+  private static String datePattern(String separator) {
+    if (separator.equals("")) {
+      return "yyyyMMdd";
+    }
+    // ' is escape character in date format, we should double it here.
+    if (separator.contains("'")) {
+      separator = separator.replace("'", "''");
+    }
+    return String.format("yyyy'%s'MM'%s'dd", separator, separator);
```

Comment:
why there is `'%s` the `'s` ?

Reply:
The separator may be more than one character.